### ISSUE TITLE
replaceFile not considered by Upload.php

### ISSUE THREAD

**Issue by g4b0** [silverstripe/silverstripe-framework#5137]
SS 3.1.18
Upload class always overwrite files, also if replaceFile is set to false. 
Changing https://github.com/silverstripe/silverstripe-framework/blob/3.1.18/filesystem/Upload.php#L161 to the code below works as expected.

```php
if (true){
```

Following my config
```yaml
---
Name: mysite
After:
  - 'framework/*'
  - 'cms/*'
---
# YAML configuration for SilverStripe
# See http://doc.silverstripe.org/framework/en/topics/configuration
# Caution: Indentation through two spaces, not tabs
SSViewer:
  theme: 'mysite'    
# replace files instead of versioning
Upload:
  replaceFile: false
# show an overwrite warning
UploadField:
  defaultConfig:
    overwriteWarning: true
```

Following my upload field
```php
$fields->addFieldToTab(
    'Root.Main', $uploadField = new UploadField(
    'FilePercorso', 'File del percorso'
    )
);
$uploadField->setFolderName('Uploads/Percorsi');
$uploadField->setAllowedExtensions(array('gpx'));
```

**Comment by tractorcow** [comment]
Could it be an issue where you can't merge falsey configs into truthey ones?

Maybe `Config::inst()->get('Upload', 'replaceFile')` is stuck on true?

**Comment by g4b0** [comment]
@tractorcow are you speaking about my project, or do you think it's a SS issue?

**Comment by JayDevlin** [comment]
It's a framework issue for the 3.0 and 3.1 branches and was fixed for 3.2.

see #3048

**Comment by g4b0** [comment]
Will it be included in 3.1 branch?

**Comment by JayDevlin** [comment]
I don't see how this issue can be fixed in 3.1 without changing the api.

Though, you can workaround this bug by setting overwriteWarning to false.

**Comment by g4b0** [comment]
Nice workaround, it works for me. Thanks
